spring-boot-cli/src/main/java/org/springframework/boot/cli/command/CommandRunner.java

```diff
@@ -185,17 +185,29 @@ public int runAndHandleErrors(String... args) {
 	}
 
 	private String[] removeDebugFlags(String[] args) {
-		List<String> rtn = new ArrayList<String>(args.length);
+		List<String> argsWithoutDebugFlags = new ArrayList<String>(args.length);
 		boolean appArgsDetected = false;
 		for (String arg : args) {
 			// Allow apps to have a -d argument
-			appArgsDetected |= "--".equals(arg);
-			if (("-d".equals(arg) || "--debug".equals(arg)) && !appArgsDetected) {
+			appArgsDetected |= isAppArg(arg);
+			if (isDebugArgTogetherWithAppArg(arg, appArgsDetected)) {
 				continue;
 			}
-			rtn.add(arg);
+			argsWithoutDebugFlags.add(arg);
 		}
-		return rtn.toArray(new String[rtn.size()]);
+		return argsWithoutDebugFlags.toArray(new String[argsWithoutDebugFlags.size()]);
```

Comment:
toArray() is a bit costly as it does copy array to new array object.
I would say creating primitive array like:
```java
String[] argsWithoutDebugFlags = new String[args.length];
```
and then copying values in proper index by maintaining running index e.g.

```java
int index = 0;
for (String arg : args) {
...
argsWithoutDebugFlags[index++] = arg;
...
}
...
return argsWithoutDebugFlags;
```